olga/analytics/tests/test_models.py

```diff
@@ -0,0 +1,309 @@
+        """
+        enthusiast_edx_installation = {
+            'latitude': 50.32,
+            'longitude': 45.11,
+            'platform_name': 'platform_name',
+            'platform_url': 'https://platform.url'
+        }
+
+        edx_installation_object = EdxInstallation.objects.first()
+        edx_installation_object.update_edx_instance_info(enthusiast_edx_installation)
+
+        extended_edx_installation_object_attributes = edx_installation_object.__dict__
+
+        self.assertDictContainsSubset(enthusiast_edx_installation, extended_edx_installation_object_attributes)
+
+
+class TestInstallationStatisticsMethods(TestCase):
+    """
+    Tests for InstallationStatistics model's core methods, that work with database.
+    """
+
+    @patch('django.utils.timezone.now')
+    def setUp(self, mock_timezone_now):  # pylint: disable=arguments-differ
+        """
+        Create five unique edx installations and corresponding statistics.
+        Create two statistics objects for acquainted edx installation (last two edx installation objects in unique list)
+
+        So we get 5 unique objects, 3 from them has only one record in installation statistics,
+        2 of them has 2 records by one for last two unique objects.
+
+        Let me show full description for clearer understanding:
+            - first edx installation object with 2017-06-01 15:30:30 statistics object by time.
+            - second object with 2017-06-01 15:30:30
+            - third object with 2017-06-01 15:30:30
+            - fourth object with 2017-06-01 15:30:30, 2017-06-04 15:30:30, 2017-06-05 15:30:30
+            - fifth object with 2017-06-01 15:30:30, 2017-06-04 15:30:30, 2017-06-05 15:30:30
+        """
+        data_created_datetimes = [
+            datetime(2017, 6, 1, 15, 30, 30),
+            datetime(2017, 6, 1, 15, 30, 30),
+            datetime(2017, 6, 2, 15, 30, 30),
+            datetime(2017, 6, 3, 15, 30, 30),
+            datetime(2017, 6, 3, 15, 30, 30),
+        ]
+
+        for data_created_datetime in data_created_datetimes:
+            mock_timezone_now.return_value = data_created_datetime
+            InstallationStatisticsFactory(data_created_datetime=data_created_datetime)
+
+        data_created_datetimes_for_acquainted_edx_installation = [
+            datetime(2017, 6, 4, 15, 30, 30),
+            datetime(2017, 6, 5, 15, 30, 30),
+        ]
+
+        for data_created_datetime in data_created_datetimes_for_acquainted_edx_installation:
+            mock_timezone_now.return_value = data_created_datetime
+
+            InstallationStatisticsFactory(
+                data_created_datetime=data_created_datetime,
+                edx_installation=InstallationStatistics.objects.all()[4].edx_installation
+            )
+
+            InstallationStatisticsFactory(
+                data_created_datetime=data_created_datetime,
+                edx_installation=InstallationStatistics.objects.all()[5].edx_installation
+            )
```

Comment:
> better create 5 dicts with data, and pass them as kwargs to the factory, like InstallationStatisticsFactory(**iterated_dict)

Then I do not need to use Factory for that at all.
If I go ahead and use dictionaries, I can put it in simple `InstallationStatistics`, that Django supports me to do.

Need to be discussed.